#2983 Prevent XSS for URLs

Patrykb0802 opened the pull request:
Added:
- XssFilter, XSSUtils for checkign if URL has an illegal character
- Changed paths in pointHierarchySLTS.jsp
- bean in spring-utils as well as strictHttpFirewall

Patrykb0802 committed:
#2983 Prevent XSS for URLs

feature/#2983_Prevent_XSS_for_URLs

Limraj committed:
#2983 Prevent XSS for URLs:
- corrected redirect for error 500 code;

Patrykb0802 committed:
#2983 Prevent XSS for URLs

Added:
- JUnit 4 test for XssUtils.java class
- Logger for org.scada_lts.web.security in log4j2.xml

- Minor changes in XssFilter.java

Limraj commented on src/org/scada_lts/web/security/XssUtils.java:
```diff
@@ -0,0 +1,38 @@
+public class XssUtils {
+
+    private XssUtils() {
+        throw new UnsupportedOperationException("Utility class");
```
This is a no good approach, we will find out in runtime. 
It's better to make a final class to highlight it in the IDE and create a private no args constructor.

Limraj commented on test/org/scada_lts/web/security/XssUtilsTest.java:
```diff
@@ -0,0 +1,53 @@
+
+    @Test
+    public void testValidate_ContainsForbiddenPatterns() {
+        assertFalse("A string containing 'javascript:' should be considered invalid.", XssUtils.validate("This contains javascript: in the middle"));
```
One test equals one case(assert).

Limraj commented on test/org/scada_lts/web/security/XssUtilsTest.java:
```diff
@@ -0,0 +1,53 @@
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class XssUtilsTest {
```
I suggest creating a Parameterized test where you provide a list of strings.

Limraj commented on webapp-resources/log4j2.xml:
```diff
@@ -286,6 +286,11 @@
         </Logger>
 
+        <!-- XSS Utils Logging-->
+        <Logger name="org.scada_lts.web.security" level="WARN">
```
See what the rest of the Loggers look like, try to use the same pattern.

Patrykb0802 committed:
#2983 Prevent XSS for URLs

Corrected tests in XssUtilsTest.java and logger

Corrected XssUtils.validate(), now it works with all query cases

Limraj committed:
#2983 Prevent XSS for URLs:
- corrected regex pattern (check pattern: param=value, param);
- added test cases to XssUtilsTest;
- renamed method from validate to validateHttpQuery in XssUtils;
- corrected implementation Filter;

Limraj committed:
#2983 Prevent XSS for URLs:
- added test cases to XssUtilsTest;
- added properties:
scadalts.security.http.query.access.denied.regex,
scadalts.security.http.query.access.granted.regex,
scadalts.security.http.query.limit
scadalts.security.http.query.xss.enabled
to env.properties
- corrected /pointHierarchy/move/

Limraj committed:
#2983 Prevent XSS for URLs:
- included variant abc= in the regex;

Patrykb0802 committed:
#2983 Prevent XSS for URLs

 - Minor changes in log4j2.xml

Patrykb0802 committed:
#2983 Prevent XSS for URLs

 - Minor changes in log4j2.xml

Patrykb0802 committed:
#1580 Improvement describe information about scada on top

- Changed way of top description being dynamically refreshed

Patrykb0802 committed:
Revert "#1580 Improvement describe information about scada on top"

This reverts commit 7c70a42bcc961cd7b6c01acfe922b33589cb38d5.

Patrykb0802 committed:
#2983 Prevent XSS for URLs

Limraj committed:
#2983 Prevent XSS for URLs:
- fixed websocket url;

Limraj committed:
#2983 Prevent XSS for URLs:
- added Logger org.springframework.security;
- set ERROR logging level for securityLoggingLevel;
- set default query limit to 3900, default URL length limit for Apache Server a is approximately 4000 characters;

Limraj committed:
#2983 Prevent XSS for URLs:
- corrected junit test config

Limraj committed:
#2983 Prevent XSS for URLs:
- commented old configuration for http-firewall

Limraj committed:
#2983 Prevent XSS for URLs:
- corrected scadalts.security.http.query.xss.enabled to scadalts.security.http.query.protect.enabled;
- added test cases to XssUtilsTest;
